### PR TITLE
Remove obsolete iron sights

### LINKED ISSUE #27827: Iron sights do nothing
**Is your feature request related to a problem? Please describe.**  
Iron sights have sight dispersion of 30. Weapons with no mods have default sight dispersion of 30. The result is that every single weapon acts as if they already have iron sights on them making the actual iron sights mod useless.

The sight dispersion value on laser sights is equally useless. But at least those have different aim speed value.

**Describe the solution you'd like**  
Increase the default sight dispersion to above 30 so that iron sights are an improvement. The "DISABLE_SIGHTS" flag sets the weapon sight dispersion to 90 so that should be the new default value.

Most guns should also spawn with iron sights as the default mod or have them as a built in mod. The weapon weight should also be reduced by 0.06 kg to account for extra weight that comes from having iron sights mod.

Weapons that have built in iron sights would also need one more sight slot added on them since they now have an unremovable sights on them.

Some weapons would also need to have sight dispersion set on them so that they would not need iron sights. For example bows and crossbows.

**Describe alternatives you've considered**  
Remove iron sights.

**Additional context**  
Add any other context (such as mock-ups, proof of concepts or screenshots) about the feature request here.

### PR DESCRIPTION
**Pull request by Hirmuolio**
#### Summary
```SUMMARY: Content "Remove obsolete iron sights"```

#### Purpose of change

Closes #27827
(well it was already closed due to no activity but mentioning it anyways)

Iron sights do not do anything ever (except due to some bug they can be used instead of offset iron sights when the weapon has blocked sights).

Iron sights set weapons to sight dispersion 30 and aiming speed 6.
All weapons have sight dispersion of 30 and aiming speed 6 by default. So the iron sights do not do anything.

#### Describe the solution

Removed iron sights recipe and spawns. Moved the item to obsolete.

#### Describe alternatives you've considered

Make default sight dispersion on weapons worse and add iron sights as default or built in mod on almost all weapons.

### CODE REVIEW

**Comment by tenmillimaster** [comment]
I have plans for the sights. They look like this:
image

iron sights should also not mount on railspace.

**Comment by tenmillimaster** [comment]
Also, your alternative solution is preferred imo, with a handful of guns having it as a default mod and most as a permanent one.

File: data/json/recipes/weapon/mods.json [hunk 1]
```diff
@@ -1,30 +1,4 @@
 [
-  {
-    "result": "improve_sights",
```
Comment (ZhilkinSerg) [inline data/json/recipes/weapon/mods.json]: Please add deleted recipe to https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/json/recipes/recipe_obsolete.json

**Comment by Hirmuolio** [comment]
Yeah I agree adding the sights to weapons would be better but it would be quite a bit of work. Especially if one wants to use the default/build in sights distinction on all weapons. I don't know how common it is for the iron sights to be removable in weapons so maybe an automated blanket built in iron sights for all weapons would be good enough with manual fixes applied later when needed.

I really don't think iron sights working with sight blocking mods is working as intended. Nothing in iron sight description would indicate that it would work. Also the code for the sight blocking is so that the `"delete": { "flags": [ "DISABLE_SIGHTS" ] }` on offset iron sights is not needed for them to work with sight blocking mods. All sight mods work with sight blocking mods.

**Comment by tenmillimaster** [comment]
My understanding is that any disable sights mod will only block *base* sights. Any subsequent sights mod would add sights back to the weapon, working around the obstruction.

That's what makes sense to me IRL too.

**Comment by Mrkeserian** [comment]
Wait, so why shouldn't iron sights mount on rails? There are boatloads of Iron sights that attach to common rail systems (Magpul MBUS for example) to serve as backup sights in case whatever battery powered optic is on the rifle goes down.

**Comment by Hirmuolio** [comment]
This isn't about the rail mounted iron sights. This is about the normal iron sights.

**Comment by tenmillimaster** [comment]
My bad: zoned out while reading the offset sights entry.

While backup sights and other sights do mount on railspace, it's usually on the receiver and in the 12 o clock position; the rail location isn't as picky.

**Comment by tenmillimaster** [comment]
This looks like intended behavior to me.

image

https://www.trijicon.com/na_en/assets/image/BNTSS-Features2.jpg

<https://news.guns.com/wp-content/uploads/2014/02/Beretta7101.jpg>

**Comment by Hirmuolio** [comment]
Normal suppressors do not block sights in CDDA. Only the homemade ones (homemade suppressor, 'solvent trap' suppressor and soda bottle silencer) block sights.

Just slightly raised aftermarket sights shouldn't work with those.

**Comment by tenmillimaster** [comment]
The presumption is that they're installed in such a way that the firearm is sight-able again, potentially even on the suppressor. 

different example:

image

**Comment by Hirmuolio** [comment]
Can we get a decision of some sort on wether iron sights should work with gun mods that block sights?

If iron sights are supposed to work with blocked sights then this thing can be closed.  
If iron sights should not work with blocked sights then this is ready as is. (Unless someone wants to go and do the considered alternative)

**Comment by kevingranade** [comment]
The current situation is better then the alternative of removing the sights. They currently don't have a impact but they should still exist and guns that happen to not have the sights at all for some reason should be even worse.

**Pull request closed without merging**